src/App.jsx

```diff
@@ -94,7 +30,9 @@ export default function AppWithProviders() {
   return (
     <LanguageProvider>
       <ViewStateProvider>
-        <App />
+        <EntriesProvider>
+          <App />
+        </EntriesProvider>
       </ViewStateProvider>
     </LanguageProvider>
   );
```

Comment:
same thing here. it's a better practice to leave only `App` within the providers and extract the logic above to another component.